csrc/tensor_view.cpp

```diff
@@ -804,8 +804,6 @@ TensorView* TensorView::rFactor(const std::vector<int64_t>& axes) {
       "Error rfactoring ",
       this,
       " its definition is either a nullptr or not a reduction.");
-  NVF_CHECK(
-      !domain()->hasRoot(), "Cannot call rfactor on the same view twice.");
```

Comment:
I wonder why you were hitting these checks

Reply:
Got it. I don't love the use of `commitLeafToLogical` since that needs to happen at definition. Instead, in the translation PR #3440 I just create the output with K innermost and use an `AxisMapping`. I think we could probably always avoid `commitLeafToLogical` in this way; we might just need to update the tests.